### PR TITLE
Remove IValitRulesProvider

### PR DESCRIPTION
**Pull request by timdeschryver**
# Info
This Pull Request is related to issue no. #154 

# Changes
- Removed `IValitRulesProvider`
- Removed `CreateValitator` extension for `IValitRulesProvider`
- Removed `IValitRulesProvider` ctor for `Validator`
- Changed `Create` in `ValitRules` to accept `IValitRules<TObject>` instead of `IEnumerable<IValitRule<TObject>>`
- Changed tests

### CODE REVIEW

File: src/Valit/ValitatorExtensions.cs [hunk 1]
```diff
@@ -5,16 +5,10 @@ namespace Valit
 {
     public static class ValitRulesProviderExtensions
```
Comment (GooRiOn) [inline src/Valit/ValitatorExtensions.cs]: Could you change the name of this class to ``ValitRulesExtensions`` since now there's no provider ? :D

**Review by GooRiOn — COMMENTED** [review]
Except this small naming thing, everything LGTM. THX :)

File: src/Valit/ValitatorExtensions.cs (hunk 1 above)
Comment (timdeschryver) [reply src/Valit/ValitatorExtensions.cs]: You're right! commit incoming...

Comment (timdeschryver) [reply src/Valit/ValitatorExtensions.cs]: Hmm the file itself is named `ValitatorExtensions`. Do you have a preference? 
I would go for `ValitRulesExtensions`.

Comment (GooRiOn) [reply src/Valit/ValitatorExtensions.cs]: I totally agree. The name should be ``ValitRulesExtensions``

**Review by GooRiOn — APPROVED** [review]
LGTM

**Pull request merged**